packages/core/src/repositories/providerApiKeysRepository.ts

```diff
@@ -0,0 +1,32 @@
+import { NotFoundError, Result } from '$core/lib'
+import { providerApiKeys, workspaces } from '$core/schema'
+import { eq, getTableColumns } from 'drizzle-orm'
+
+import Repository from './repository'
+
+export class ProviderApiKeysRepository extends Repository {
+  get scope() {
+    return this.db
+      .select(getTableColumns(providerApiKeys))
+      .from(providerApiKeys)
+      .innerJoin(workspaces, eq(workspaces.id, providerApiKeys.workspaceId))
+      .as('providerApiKeysScope')
+  }
+
+  async find(id: number) {
+    const result = await this.db
+      .select()
+      .from(this.scope)
+      .where(eq(this.scope.id, id))
+    if (!result.length) {
+      return Result.error(new NotFoundError('ProviderApiKey not found'))
```

Comment:
Be aware that in the server component you have to handle this with [notFound()](https://nextjs.org/docs/pages/api-reference/functions/get-server-side-props#notfound) to show users the not found error boundary. Otherwise this is treated like an exception